From 9b69a026bc13c6c6613390286432e3c4ae258acb Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Wed, 2 Jun 2021 14:17:13 -0400
Subject: [PATCH 42/79] waylandsink: Fix for missing initial configure

We were doing our initial "empty" commit on the subsurface instead of the
toplevel surface. As an incidence, we should not have received a configure
event ever, not just on mutter. This fixes the following warning when using
mutter compositor (aka gnome-shell):

  waylandsink wlwindow.c:304:gst_wl_window_new_toplevel: The compositor did not send configure event.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/-/merge_requests/2299>
(cherry picked from commit 56b56e43f30a6aecf39cc0b02688a88d6130c947)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 ext/wayland/wlwindow.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/wayland/wlwindow.c b/ext/wayland/wlwindow.c
index ff66166c1..b10c646e3 100644
--- a/ext/wayland/wlwindow.c
+++ b/ext/wayland/wlwindow.c
@@ -291,7 +291,7 @@ gst_wl_window_new_toplevel (GstWlDisplay * display, const GstVideoInfo * info,
 
     /* Finally, commit the xdg_surface state as toplevel */
     window->configured = FALSE;
-    wl_surface_commit (window->video_surface);
+    wl_surface_commit (window->area_surface);
     wl_display_flush (display->display);
 
     g_mutex_lock (&window->configure_mutex);
-- 
2.20.1

